### PR TITLE
Carnix

### PR DESCRIPTION
**Pull request by grahamc**

### CODE REVIEW

File: default.nix [hunk 1]
```diff
@@ -2,41 +2,33 @@
...
-  '';
-
-  # Darwin fails to build doctests with:
-  # dyld: Symbol not found __ZTIN4llvm2cl18GenericOptionValueE
-  # see: https://github.com/NixOS/nixpkgs/pull/49839
-  doCheck = !pkgs.stdenv.isDarwin;
+((pkgs.callPackage ./Cargo.nix {
+  cratesIO = pkgs.callPackage ./crates-io.nix {};
+}).lorri {
+}).override {
```
Comment (tilpner) [inline default.nix]: Could be single line to make empty set more obvious

**Comment by Profpatsch** [comment]
That’s a *lot* of lines for a simple lockfile. Is there a deeper reason why we have to check in three different generated files to get carnix support?

**Comment by grahamc** [comment]
The Cargo.nix must live in the root of the project, as the generated file always uses `./.` as the source, and I'm not interested in spending a bunch of time fixing that to avoid having the file in the root.

**Comment by Profpatsch** [comment]
Great! If we add Cachix now, we can get the build times down considerably.

Should I squash-merge?

**Comment by grahamc** [comment]
Sure.

**Pull request merged**